ts/features/fastLogin/saga/tokenRefreshSaga.ts

```diff
@@ -0,0 +1,133 @@
+import { SagaIterator } from "redux-saga";
+import { call, delay, put, takeLatest } from "typed-redux-saga/macro";
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { ValidationError } from "io-ts";
+import {
+  refreshSessionToken,
+  sessionExpired
+} from "../../../store/actions/authentication";
+import { SessionToken } from "../../../types/SessionToken";
+import { startApplicationInitialization } from "../../../store/actions/application";
+import {
+  createFastLoginClient,
+  createNonceClient,
+  performFastLogin,
+  performGetNonce
+} from "../backend";
+import { apiUrlPrefix } from "../../../config";
+import { SagaCallReturnType } from "../../../types/utils";
+import { LollipopConfig } from "../../lollipop";
+import { getKeyInfo } from "../../lollipop/saga";
+import {
+  FastLoginResponse,
+  FastLoginResponseType,
+  NonceBaseResponseType,
+  NonceResponse
+} from "../backend/mockedFunctionsAndTypes";
+
+export function* watchTokenRefreshSaga(): SagaIterator {
+  yield* takeLatest(refreshSessionToken.request, doRefreshTokenSaga);
+}
+
+type RequestStateType = {
+  counter: number;
+  success: boolean;
```

Comment:
I wouldn't use a simple `boolean` to  map the "`success`". I would instead use a more explicit "`status`" field like for example `"in-progress" | "succeded" | "failed"` (for example) and I would use a discriminated unions to map the entire `RequestStateType` what do you think?

Reply:
Or if you prefer to leave it the way it is, we could just add a comment a track it on with a JIRA ticket.

Reply:
I think the first solution you proposed, more explicit, is generally the best. I'm going to edit :+1:

Reply:
Ops.. the second comment was for the `MAX_RETRIES` part... I put there another comment ;)